packages/components/src/components/checkbox/checkbox.tsx

```diff
@@ -0,0 +1,215 @@
+import {
+  AttachInternals,
+  Component,
+  Element,
+  Event,
+  type EventEmitter,
+  h,
+  type JSX,
+  Listen,
+  Prop,
+  Watch,
+} from '@stencil/core';
+import {
+  AllowedTypes,
+  applyConstructableStylesheetStyles,
+  attachComponentCss,
+  FORM_STATES,
+  getPrefixedTagNames,
+  hasPropValueChanged,
+  isDisabledOrLoading,
+  THEMES,
+  validateProps,
+} from '../../utils';
+import { type BreakpointCustomizable, type PropTypes, type Theme } from '../../types';
+import { getComponentCss } from './checkbox-styles';
+import type { CheckboxState, CheckboxUpdateEventDetail } from './checkbox-utils';
+import { messageId, StateMessage } from '../common/state-message/state-message';
+import { descriptionId, Label } from '../common/label/label';
+import { LoadingMessage } from '../common/loading-message/loading-message';
+import { ControllerHost, InitialLoadingController } from '../../controllers';
+import { getCheckboxRadioButtonSafariRenderingFix } from '../../utils/form/applyCheckboxRadioButtonSafariRenderingFix';
+import { getSlottedAnchorStyles } from '../../styles';
+
+const propTypes: PropTypes<typeof Checkbox> = {
+  label: AllowedTypes.string,
+  name: AllowedTypes.string,
+  value: AllowedTypes.string,
+  required: AllowedTypes.boolean,
+  disabled: AllowedTypes.boolean,
+  indeterminate: AllowedTypes.boolean,
+  checked: AllowedTypes.boolean,
+  form: AllowedTypes.string,
+  state: AllowedTypes.oneOf<CheckboxState>(FORM_STATES),
+  message: AllowedTypes.string,
+  hideLabel: AllowedTypes.breakpoint('boolean'),
+  loading: AllowedTypes.boolean,
+  theme: AllowedTypes.oneOf<Theme>(THEMES),
+};
+
+/**
+ * @slot {"name": "label", "description": "Shows a label. Only [phrasing content](https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/Content_categories#Phrasing_content) is allowed." }
+ * @slot {"name": "message", "description": "Shows a state message. Only [phrasing content](https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/Content_categories#Phrasing_content) is allowed." }
+ */
+@Component({
+  tag: 'p-checkbox',
+  shadow: { delegatesFocus: true },
+  formAssociated: true,
+})
+export class Checkbox {
+  @Element() public host!: HTMLElement;
+
+  /** The name of the checkbox. */
+  @Prop({ reflect: true }) public name?: string = '';
+  // The "name" property is reflected as an attribute to ensure compatibility with native form submission.
+  // In the React wrapper, all props are synced as properties on the element ref, so reflecting "name" as an attribute ensures it is properly handled in the form submission process.
+
+  /** Marks the checkbox as required. */
+  @Prop() public required?: boolean = false;
+
+  /** Marks the checkbox as disabled. */
+  @Prop() public disabled?: boolean = false;
+
+  /** Marks the checkbox as indeterminate. */
+  @Prop() public indeterminate?: boolean = false;
+
+  /** Marks the checkbox as pre-selected (checked) on initial load. */
+  @Prop() public checked?: boolean = false;
```

Comment:
This prop should be mutable and reflect the current checkbox checked state. When used as an attribute it will determine whether this checkbox is checked by default (when the page loads).